python-json-rbac/core.py

```
@@ -0,0 +1,125 @@
+            algorithms=[ALGORITHM],
+            options={"verify_exp": True, "verify_nbf": True, "verify_iat": True},
+        )
+
+        # --- Enforce required claims ------------------------------------------
+        if "sub" not in payload or "role" not in payload:
+            raise JWTClaimsError("Missing required claims")
+
+        return payload
+
+    except ExpiredSignatureError:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Token expired")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)